pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -341,22 +339,12 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 					adsLog.Debugf("ADS:EDS: ACK %s %s %s %s", peerAddr, con.ConID, discReq.VersionInfo, discReq.ResponseNonce)
 					if discReq.ResponseNonce != "" {
 						con.mu.Lock()
-						edsClusterMutex.RLock()
 						con.EndpointNonceAcked = discReq.ResponseNonce
-						if len(edsClusters) != 0 {
-							con.EndpointPercent = int((float64(len(clusters)) / float64(len(edsClusters))) * float64(100))
-						}
-						edsClusterMutex.RUnlock()
 						con.mu.Unlock()
 					}
 					continue
 				}
 
-				previous := sets.NewSet(con.Clusters...)
```

Comment:
I think sets was added just for this purpose - Do you want to retain them even if they are not used? I am OK to leave them like that as they could be used in else where in future.